bisectorfunctions.py

```diff
@@ -0,0 +1,27 @@
+import copy
+import functools
+import logging
+import subprocess
+from typing import Optional
+
+from ccbuilder import Builder, Commit
+
+import checker
+import utils
+
+
+def test_bisection(
+    new_commit: Commit, chkrncse: tuple[checker.Checker, utils.Case], bldr: Builder
```

Comment:
why is the checker and case bundled together?

Reply:
As the checking function is not standalone, we need the checker to run the test. The only way we can pass additional thinks to our test function is by making it part of the `state`.